packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -549,7 +565,22 @@ export class FileSystem extends FileSystemBase {
       defaultPort: ec2.Port.tcp(FileSystem.DEFAULT_PORT),
     });
 
-    const subnets = props.vpc.selectSubnets(props.vpcSubnets ?? { onePerAz: true });
+    // When oneZone is specified, to avoid deployment failure, mountTarget should also be created only in the specified AZ.
+    let subnetSelection: ec2.SubnetSelection;
+    if (props.oneZone) {
+      subnetSelection = {
+        availabilityZones: [oneZoneAzName],
+      };
+      if (props.vpcSubnets) {
+        subnetSelection = {
+          ...props.vpcSubnets,
+          ...subnetSelection,
+        };
```

Comment:
Not sure about this, what happens if the user specifies subnets from different AZs?

Reply:
Probably, the return value from props.vpc.selectSubnets(subnetSelection) is an empty array, and the mountTarget is not created.

Should it be considered an error when the return value of selectSubnets is an empty array with oneZone=true?

Reply:
Given: 
* [Amazon EFS One Zone file systems support only a single mount target which is located in the same Availability Zone as the file system. You cannot add additional mount targets.](https://docs.aws.amazon.com/efs/latest/ug/mounting-one-zone.html)
* `selectSubnets` [Requires that at least one subnet is matched, throws a descriptive error message otherwise.](https://github.com/aws/aws-cdk/blob/ca91626327ac4a138adc16026d951f21fadf86a3/packages/aws-cdk-lib/aws-ec2/lib/vpc.ts#L129)

I think that the best option here is to validate that the users can specify either `oneZone` or `vpcSubnets`, not both, to prevent deployment errors and change the implementation accordingly.